raster/r.report/format.c

```diff
@@ -73,3 +73,93 @@ int format_double(double v, char *buf, int n, int dp)
 
     return 0;
 }
+
+void compute_unit_format(int unit1, int unit2, enum OutputFormat format)
+{
+    int i, ns, len;
+    char num[100];
+    int need_format;
+
+    /* examine units, determine output format */
+    for (i = unit1; i <= unit2; i++) {
+        if (format == PLAIN) {
+            need_format = 1;
+        }
+        else {
+            need_format = 0;
+        }
+        unit[i].label[0] = "";
+        unit[i].label[1] = "";
+
+        switch (unit[i].type) {
+        case CELL_COUNTS:
+            unit[i].label[0] = " cell";
+            unit[i].label[1] = "count";
+
+            if (need_format) {
+                need_format = 0;
+                unit[i].len = 5;
+                ns = 0;
+                sprintf(num, "%ld", count_sum(&ns, -1));
```

Comment:
I know this is only old code moved to new place (and sorry for nitpicking). For safety reasons (and possibly future portability, in this case this argument is perhaps the prime argument) it is preferable to use `snprintf()`.
```suggestion
                snprintf(num, sizeof(num), "%ld", count_sum(&ns, -1));
```

Reply:
Updated.